src/components/UserRequests/requests.js

```diff
@@ -0,0 +1,385 @@
+  }
+
+  setShow = (value) => {
+    this.setState({
+      showModal: value,
+    });
+  };
+
+  onChange = (e) => {
+    this.setState({
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  handleClose = () => {
+    this.setShow(false);
+    this.setState({
+      quantity: 1,
+      location: "",
+      blood_group: "",
+      priority: "",
+      to_edit_id: null,
+    });
+  };
+
+  handleShow = (req) => {
+    this.setState({
+      to_edit_id: req.id,
+      quantity: req.quantity,
+      location: req.location,
+      blood_group: req.blood_group,
+      priority: req.priority,
+    });
+
+    this.setShow(true);
+  };
+
+  editRequestSubmit = (e) => {
+    e.preventDefault();
+    console.log("here");
+
+    const id = this.state.to_edit_id;
+    if (id) {
+      const data = {
+        quantity: this.state.quantity,
+        location: this.state.location,
+        priority: this.state.priority,
+        blood_group: this.state.blood_group,
+      };
+      this.editRequest(id, data);
+      this.handleClose();
+    }
+  };
+
+  async editRequest(id, data) {
+    const config = {
+      method: "patch",
+      url: BASE_URL + EDIT_REQUEST + id + "/",
+      data: data,
+    };
+    axios(config)
+      .then((res) => {
+        const data = res.data;
+        let pR = [...this.state.pendingRequests]
+        let objIndex = pR.findIndex((obj => obj.id === data.id));
+        pR[objIndex] = data
+        this.setState({
+          pendingRequests:pR
+        })
+        alert("Request Edited");
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+
+  async deleteRequest(id) {
+    swal({
+      title: "Are you sure?",
+      text: "Once deleted, you will not be able to recover this",
+      icon: "warning",
+      buttons: true,
+      dangerMode: true,
+    }).then((willDelete) => {
+      if (willDelete) {
+        const url = BASE_URL + DELETE_REQUEST + id + "/";
+        const config = {
+          method: "delete",
+          url: url,
+        };
+        axios(config)
+          .then((res) => {
+            console.log(res)
+            if(res.status === 204)
+            {
+              alert("Request Deleted");
+            }
+            let pR = [...this.state.pendingRequests]
+            let objIndex = pR.findIndex((obj => obj.id === id));
+            pR.splice(objIndex,1)
+            this.setState({
+              pendingRequests:pR
+            })
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      }
+    });
+  }
+
+  calculateRequestStats = (data) => {
+    let completedRequests = data.filter((obj) => {
+      return obj.is_complete === true;
+    });
+    data = data.filter((el) => !completedRequests.includes(el));
+    let approvedRequests = data.filter((obj) => {
+      return obj.is_approved === true;
+    });
+    let pendingRequests = data.filter((el) => !approvedRequests.includes(el));
```

Comment:
why don't we just only pull the requests that we need and filter the requests on server-side instead?